stl/inc/yvals_core.h

```diff
@@ -855,6 +855,42 @@
 
 // next warning number: STL4027
 
+// P0619R4 remove deprecated features from C++20
+#ifndef _HAS_REMOVED_FEATURES_CXX20
+#define _HAS_REMOVED_FEATURES_CXX20 (!_HAS_CXX20)
+#endif // _HAS_REMOVED_FEATURES_CXX20
+
+#ifndef _HAS_DEPRECATED_TYPETRAITS
+#define _HAS_DEPRECATED_TYPETRAITS (!_HAS_CXX20 || _HAS_REMOVED_FEATURES_CXX20)
+#endif // _HAS_DEPRECATED_TYPETRAITS
+
+#ifndef _HAS_DEPRECATED_SHARED_PTR_UNIQUE
+#define _HAS_DEPRECATED_SHARED_PTR_UNIQUE (!_HAS_CXX20 || _HAS_REMOVED_FEATURES_CXX20)
+#endif // _HAS_DEPRECATED_SHARED_PTR_UNIQUE
+
+#ifndef _HAS_DEPRECATED_NEGATE
+#define _HAS_DEPRECATED_NEGATE (!_HAS_CXX20 || _HAS_REMOVED_FEATURES_CXX20)
+#endif // _HAS_DEPRECATED_NEGATE
+
+#ifndef _HAS_DEPRECATED_ADAPTOR_TYPEDEFS
+#define _HAS_DEPRECATED_ADAPTOR_TYPEDEFS (!_HAS_CXX20 || _HAS_REMOVED_FEATURES_CXX20)
+#endif // _HAS_DEPRECATED_ADAPTOR_TYPEDEFS
+
+#if !_HAS_DEPRECATED_ADAPTOR_TYPEDEFS

```

Comment:
Style: We conventionally prefer to avoid negated tests when it's possible. Here, you have an `#if/#else`, so I recommend inverting them.

Reply:
Done